presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -57,8 +59,8 @@
     private static final int MAX_CATALOG_NAME_LENGTH = 128;
     private static final int MAX_SCHEMA_NAME_LENGTH = 128;
     private static final int MAX_FUNCTION_NAME_LENGTH = 256;
-    private static final int MAX_PARAMETER_TYPES_LENGTH = 500;
-    private static final int MAX_RETURN_TYPE_LENGTH = 256;
+    private static final int MAX_PARAMETER_TYPES_LENGTH = 20000;
```

Comment:
You can just use text rather than medium text if this is the upper limit?

Reply:
The max length is for the parameter type list, while the `mediumtext` is for `parameters`, which has a greater length than the parameter type list.
For example, for a parameter type list `int,varchar`, the `parameter` may be:
```
[ {
  "name" : "p1",
  "type" : "int"
}, {
  "name" : "p2",
  "type" : "varchar"
} ]
```

The 1) length of the parameter names, 2) the number of parameters, 3) the JSON-encoding characters may expand the total size.

I think we should also place limits on the parameter name length and the parameter counts. That can help us maintain the total length of `parameters` within text, although only by a small margin.